R/build_xgboost.R

```diff
@@ -348,10 +348,18 @@ augment.xgboost_binary <- function(x, data = NULL, newdata = NULL, ...) {
       data
     }
 
+    # copy ret_data to add response_col if it doesn't exist.
+    # it's needed for model.matrix function
+    md_frame <- ret_data
```

Comment:
Looks like this md_frame is just a temporary function, but it made me thing what "md" stands for, which I'm still not sure.
Could you use more obvious name, or maybe just overwrite ret_data?